### DISCUSSION TITLE
Session callback returning 304 status code.

### DISCUSSION THREAD

**Discussion by eiman-waheed** [nextauthjs/next-auth#8525] [Help]
### Environment

System:
    OS: macOS 13.4
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 230.96 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.12.0 - ~/.nvm/versions/node/v18.12.0/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 8.19.2 - ~/.nvm/versions/node/v18.12.0/bin/npm
  Browsers:
    Chrome: 116.0.5845.179
    Safari: 16.5

### Reproduction URL

https://www.google.com/

### Describe the issue

Use case explained:
We have a microservice in which we login and it sets http only cookies, then there is another microservice which is using next-auth session callback (api/auth/session) to login the user without requiring the user to enter credentials again. Session callback is triggered before the session token is craeted and gives 304 status code (not modified).

### How to reproduce

You can use cross site platforms and below are the steps to reproduce issue:

- Enter valid credentials on one site which will set cookies.
- Redirect to another site which will give an error, it should have been logged in with the same session user used above but it will not because of the session not maintained (api/auth/session will give 304 status code).

### Expected behavior

Session callback (api/auth/session) should set the user session via cookies hence should be triggered (giving 200 status code) and return the logged in user credentials.

**Comment by jaosn60810** [comment]
I have same issue
